tests/library/browsertype-connect.spec.ts

```diff
@@ -765,6 +765,20 @@ for (const kind of ['launchServer', 'run-server'] as const) {
       await browser.close();
     });
 
+    test('should connect over http proxy', {
+      annotation: { type: 'issue', description: 'https://github.com/microsoft/playwright/issues/33894' },
+    }, async ({ connect, startRemoteServer, proxyServer }) => {
+      const remoteServer = await startRemoteServer(kind);
+      const url = new URL(remoteServer.wsEndpoint());
+      proxyServer.forwardTo(+url.port, { allowConnectRequests: true });
+      const browser = await connect(`http://some.random.host.does.not.exist:1337`, {
+        proxy: { server: `localhost:${proxyServer.PORT}` },
+      });
+      const page = await browser.newPage();
+      expect(await page.evaluate('11 * 11')).toBe(121);
+      await browser.close();
```

Comment:
We want to check connectHosts there otherwise it would also pass before your change?

Reply:
It would not, because `some.random.host.does.not.exist` does not resolve. However, I can check hosts just in case.